### PR TITLE
Java 8 update

### PR DESCRIPTION
**Pull request by Tomahna**
Changing a bunch of methods by using streams instead of for/if loops

Replacing custom Filter interface by java.util.Predicate

Returning a List<File> instead of File[] when possible

### CODE REVIEW

File: phoenicis-core/src/main/java/com/playonlinux/containers/DefaultContainersManager.java [hunk 1]
```diff
@@ -50,17 +50,18 @@
...
+    public List<Container<? extends ContainerEntity>> getContainers() {
         return new ArrayList<>(containers);
     }
 
-    public void update(File[] files) {
+    public void update(List<File> files) {
         containers.clear();
+        
         for(File file: files) {
-            final ConfigFile containerConfigFile = new CompatibleConfigFileFormat(new File(file, "playonlinux.cfg"));
```
Comment (qparis) [inline phoenicis-core/src/main/java/com/playonlinux/containers/DefaultContainersManager.java]: Please keep those final

File: phoenicis-core/src/main/java/com/playonlinux/containers/DefaultContainersManager.java [hunk 2]
```diff
@@ -80,10 +81,8 @@ public void shutdown() {
 
     @Override
     public void init() throws ServiceInitializationException {
-        final File containersDirectory = playOnLinuxContext.makeContainersPath();
+        File containersDirectory = playOnLinuxContext.makeContainersPath();
```
Comment (qparis) [inline phoenicis-core/src/main/java/com/playonlinux/containers/DefaultContainersManager.java]: Please keep those final

File: phoenicis-core/src/test/java/com/playonlinux/filesystem/DirectoryWatcherFilesTest.java [hunk 3]
```diff
@@ -63,13 +65,13 @@ public void testObservableDirectory_dontChangeAnything_ObservableOnlyNotifiedOnc
 
 
         try (DirectoryWatcherFiles directoryWatcherFiles = new DirectoryWatcherFiles(mockExecutorService, temporaryDirectory, CHECK_INTERVAL)) {
-            final Consumer<File[]> mockConsumer = mock(Consumer.class);
+            Consumer<List<File>> mockConsumer = mock(Consumer.class);
```
Comment (qparis) [inline phoenicis-core/src/test/java/com/playonlinux/filesystem/DirectoryWatcherFilesTest.java]: Please keep those final

**Comment by Tomahna** [comment]
The keyword final is useless in all of these examples. I'll put it back but final should still be used only when necessary.

**Comment by qparis** [comment]
It is a matter of convention. Please keep the same conventions
In POL code, we've chosen to always use final, unless we need to modify the fields later. 

http://stackoverflow.com/questions/154314/when-should-one-use-final-for-method-parameters-and-local-variables

**Comment by qparis** [comment]
ok to test

**Comment by Veske** [comment]
Looks good, streams are nice.

**Pull request merged**